presto-mysql/src/test/java/io/prestosql/plugin/mysql/TestMySqlTypeMapping.java

```diff
@@ -484,6 +485,35 @@ public void testDouble()
                 .execute(getQueryRunner(), mysqlCreateAndInsert("tpch.mysql_test_double"));
     }
 
+    @Test(dataProvider = "testUnsignedTypesProvider")
+    public void testUnsignedTypes(String type, String prestoType, String value)
+    {
+        try (TestTable testTable = new TestTable(
+                mysqlServer::execute,
+                "tpch.test_unsigned_types",
+                format("(d_col %s unsigned)", type),
+                singletonList(value))) {
+            assertQuery(
+                    "SELECT d_col FROM " + testTable.getName(),
+                    "VALUES " + value);
+            assertQuery(
+                    sessionWithDecimalMappingAllowOverflow(UNNECESSARY, 8),
+                    format("SELECT column_name, data_type FROM information_schema.columns WHERE table_schema = 'tpch' AND table_schema||'.'||table_name = '%s'", testTable.getName()),
+                    format("VALUES ('d_col', '%s')", prestoType));
+        }
+    }
+
+    @DataProvider
+    public static Object[][] testUnsignedTypesProvider()
+    {
+        return new Object[][] {
+                {"tinyint", "smallint", "255"},
+                {"smallint", "integer", "65535"},
+                {"integer", "bigint", "4294967295"},
+                {"bigint", "decimal(20,0)", "18446744073709551615"},
```

Comment:
Did you try defining a `DataType` for them?

This could perhaps look like this:

```
private static DataType<Short> mysqlUnsignedTinyint()
{
    return dataType("tinyint unsigned", SMALLINT, Object::toString);
}
```

see `testJson` for example. In your test you would use `mysqlCreateAndInsert` runner only, as Presto obviously cannot create these types.

Reply:
Thanks for the information, it's helpful!
I will update the unit tests with DateType.

Reply:
Yeah exactly. You can start from this one:
```
    @Test
    public void testUnsignedTypes()
    {
        DataType<Short> mysqlUnsignedTinyInt = DataType.dataType("TINYINT UNSIGNED", SmallintType.SMALLINT, Objects::toString);
        DataType<Integer> mysqlUnsignedSmallInt = DataType.dataType("SMALLINT UNSIGNED", IntegerType.INTEGER, Objects::toString);

        DataTypeTest.create()
                .addRoundTrip(mysqlUnsignedTinyInt, (short) 255)
                .addRoundTrip(mysqlUnsignedSmallInt, 65535)
                .execute(getQueryRunner(), mysqlCreateAndInsert("tpch.mysql_test_unsigned_types"));
    }
```

Reply:
@chhsiao90, @losipiuk 's suggestion is even better.
if we do not need to reuse the `DataType` definitions, we can declare them inline in the test body.